diff --git a/test/suites/multi_core/CMakeLists.txt b/test/suites/multi_core/CMakeLists.txt
index 696e3f5..26d67e5 100644
--- a/test/suites/multi_core/CMakeLists.txt
+++ b/test/suites/multi_core/CMakeLists.txt
@@ -1,5 +1,6 @@
 #-------------------------------------------------------------------------------
 # Copyright (c) 2020-2021, Arm Limited. All rights reserved.
+# Copyright (c) 2021, Infineon Technologies AG. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -10,6 +11,9 @@ cmake_policy(SET CMP0079 NEW)
 if (NOT TFM_MULTI_CORE_TOPOLOGY)
     return()
 endif()
+if (NOT TEST_NS)
+    return()
+endif()
 
 add_library(tfm_test_suite_multi_core_ns STATIC EXCLUDE_FROM_ALL)
 
